src/form.js

```diff
@@ -308,6 +309,11 @@ define([
                         meta = null;
                     }
                     this.knownInstances[attrs.id] = attrs.src;
+                } else if (meta.attributes.id !== attrs.id) {
+                    // support renaming when there is no reference
+                    if (meta.internal) {
+                        meta.attributes.id = attrs.id;
+                    }
                 }
             } else if (attrs.id) {
                 // attrs has no src, find by id
```

Comment:
Seems like another possible fix for the bug would be to amend this branch to recognize and mark as not internal any instance that contains child nodes (but no src attribute). We may also need to somehow track those instances in the `knownInstances` mapping.